### PR TITLE
[NFC][Partitioner] Code refactoring

### PR DESCRIPTION
**Pull request by beicy**
Summary:
This is a follow up code refactoring after #3062.  Basically, we use updateGraphMemInfoByAddingNode to update the memory usage of a node set to eliminate some duplicated calculation. 

Documentation:

[Optional Fixes #issue]

Test Plan:
ninja test

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: lib/Partitioner/Partitioner.cpp [hunk 1]
```diff
@@ -423,19 +423,21 @@ void Partitioner::partitionsAdjust(NodeToFunctionMap &partitions,
     gain = false;
     for (FunctionToNodesMapTy::iterator it = nodesSet.begin();
          it != nodesSet.end(); ++it) {
-      NodesSetTy nSet = (*it).second;
-      std::vector<Node *> outUsers = getOutUsersWithOnePredecessor(nSet);
+      NodesSetTy *curSet = &((*it).second);
```
Comment (opti-mix) [inline lib/Partitioner/Partitioner.cpp]: Should it rather be `NodesSetTy &curSet`? Or do you plan to change the value of curSet pointer?

Comment (beicy) [reply lib/Partitioner/Partitioner.cpp]: Yes, the value maybe changed by inserting new node into this set.

Comment (opti-mix) [reply lib/Partitioner/Partitioner.cpp]: I'm not asking about the content of `NodesSetTy`. I'm saying if you assign to the NodesSetTy pointer only once, like in this line, you can replace it with a reference. But if you need to assign to the pointer multiple times, you really need a pointer.

Comment (beicy) [reply lib/Partitioner/Partitioner.cpp]: You are right!  No need to use the pointer now. Thanks!

**Pull request closed without merging**